character-sfx.js

```diff
@@ -90,11 +90,19 @@ class CharacterSfx {
       };
     }
   }
-  update(timestamp, timeDiffS) {
+  update(timestamp, timeDiffS, actions = []) {

```

Comment:
Don't pass actions into the update method. Whatever you need should be attached in the constructor and queried here.